mypy/newsemanal/semanal.py

```diff
@@ -1880,17 +1888,32 @@ def visit_assignment_stmt(self, s: AssignmentStmt) -> None:
         self.check_and_set_up_type_alias(s)
         self.newtype_analyzer.process_newtype_declaration(s)
         self.process_typevar_declaration(s)
-        if isinstance(s.rvalue, CallExpr) and isinstance(s.rvalue.analyzed, NamedTupleExpr):
-            pass  # This is a valid and analyzed named tuple definition, nothing to do here.
-        else:
-            self.named_tuple_analyzer.process_namedtuple_definition(s, self.is_func_scope())
+        self.analyze_namedtuple_assign(s)
         self.typed_dict_analyzer.process_typeddict_definition(s, self.is_func_scope())
         self.enum_call_analyzer.process_enum_call(s, self.is_func_scope())
         self.store_final_status(s)
         if not s.type:
             self.process_module_assignment(s.lvalues, s.rvalue, s)
         self.process__all__(s)
 
+    def analyze_namedtuple_assign(self, s: AssignmentStmt) -> None:
+        """Check if s defines a namedtuple."""
+        if isinstance(s.rvalue, CallExpr) and isinstance(s.rvalue.analyzed, NamedTupleExpr):
+            return  # This is a valid and analyzed named tuple definition, nothing to do here.
+        if len(s.lvalues) != 1 or not isinstance(s.lvalues[0], NameExpr):
+            return
+        lvalue = s.lvalues[0]
+        name = lvalue.name
+        is_named_tuple, info = self.named_tuple_analyzer.check_namedtuple(s.rvalue, name,
+                                                                          self.is_func_scope())
+        if not is_named_tuple:
+            return
+        # Yes, it's a valid namedtuple, but defer if it is not ready.
+        if not info:
+            self.add_symbol(name, PlaceholderNode(self.qualified_name(name), lvalue, True),
```

Comment:
`mark_incomplete` adds a `PlaceholderNode`. The logic is a bit delicate so it's important that all additions go through the same implementation.